doc/tut/group.xml

```diff
@@ -168,9 +168,9 @@ action domain of <C>norm</C>. (It only happens that both are subsets of the
 natural numbers.) We can now form  images and preimages under the natural
 homomorphism. The set of  preimages of an element under  <C>hom</C> is a coset
 modulo <C>elab</C>.
-We use the function <Ref Func="PreImages" BookName="ref"/> here because
```

Comment:
I think this can just be left as `PreImages`, in general in documentation we don't encourage people to use `NC` methods of functions. Of course at the moment it doesn't make any difference.

Reply:
@ChrisJefferson Is this rule formulated somewhere in the documentation?

(If we do not want to encourage using `NC` variants in the documentation then this pull request should not change examples from the manual to use `PreImagesRepresentativeNC` instead of `PreImagesRepresentative`.)

I would agree that the `NC` variants are mainly intended to be used inside functions where it is clear from the context that the tests are not necessary. On the other hand, one can argue that the same holds for an interactive GAP session, for example, if one wants to compute a preimage of an element that was just constructed as an element of the range of the mapping in question.

Reply:
@ThomasBreuer Looking around when defining function we often define the NC and non-NC version at the same time, but when we reference we often only reference one.

I do think if we are mentioning `PreImagesNC` we should say what the difference is -- I know someone could follow the links to PreImages and PreImagesNC, but it is a bit strange here to see both mentioned.

In general it is probably worth thinking about if we want to use `PreImagesRepresentative` or `PreImagesRepresentativeNC` in most examples. I am not sure in general how much slower `PreImagesRepresentative` is going to be, so which we should be encouraging as a default -- but I would hope `PreImagesRepresentative`, because it is safer for users who are beginners.

Reply:
@ChrisJefferson Thanks for your thoughts.

Currently I find one statement about `NC` variants in the Tutorial Manual, which talks about `Sub<something>` vs. `Sub<something>NC`, and one statement about `Sub<struct>NC` in the Reference Manual Section "Constructing Subdomains"
I think there should be a *general* statement about `Func` vs. `FuncNC` (in the Tutorial as well as in the Reference Manual), saying that the latter omits some argument checks and that the documentation of the individual functions will explain which checks are omitted.
Some example that illustrates the difference should be shown in the Tutorial: When one creates a subgroup of a group then calling `SubgroupNC` instead of `Subgroup` will avoid membership tests for the subgroup generators; in the case of a permutation group, this may mean that the computation of a stabilizer chain for the big group can be avoided, which would hopefully be not expensive; for a matrix group such as the Baby Monster, calling `SubgroupNC` will be the only reasonable way to create a subgroup, since asking for membership will be in general hopeless.

In the Reference Manual, `Func` and `FuncNC` should be documented in the same ManSection, and then cross-references need to mention only the non-`NC` variant.

Concerning the use of `NC` variants in manual examples, I see two possible strategies. Either we use only the non-`NC` variants, or we use the `NC` variants in all those cases where the context admits this (perhaps with a textual comment why the `NC` variant may be called in examples in the Tutorial and in those cases where this is not obvious).
I would expect that the differences in runtime are negligible in manual examples, and if this is not the case for some example then this example deserves a comment about this fact.

Reply:
I agree that it is not a good idea to mention NC variants in the tutorial manual, and will make the necessary changes.  The more general comments are very interesting but are beyond the scope of this PR.

Reply:
Removed NC references in doc/tut/group.xml and doc/tut/algvspc.xml.  Had problems trying to push these - no idea why - so used --force.  Hope that does not cause any problems.